### ISSUE TITLE
ArrowNotImplementedError when writing to_parquet()

### ISSUE THREAD

**Issue by drahnreb** [scikit-hep/awkward#678]
A valid array raises `ArrowNotImplementedError` with the following Error during write of a parquet file:
```python
>>> import awkward as ak
>>> import pyarrow as pa
>>> (ak.__version__, pa.__version__)
('1.0.2', '2.0.0')
>>> ak.is_valid(array)
True
>>> ak.to_parquet(array, 'out.parquet')
ArrowNotImplementedError: Lists with non-zero length null components are not supported
```
Not sure if it is an upstream bug. Maybe related to [ARROW-1644](https://issues.apache.org/jira/browse/ARROW-1644)?
It does seem to be related to how the array is constructed though.

A round trip over pandas works while a pure arrow export raises the same error:
```python
>>> pa.Table.to_pandas(ak.to_arrow_table(array)).to_parquet('out.parquet')
>>> ak.is_valid(ak.from_parquet('out.parquet'))
True
>>> pa.parquet.write_table(ak.to_arrow_table(array), 'out.parquet')
/lib/python3.8/site-packages/pyarrow/parquet.py in write_table(self, table, row_group_size)
    589             raise ValueError(msg)
    590 
--> 591         self.writer.write_table(table, row_group_size=row_group_size)
ArrowNotImplementedError: Lists with non-zero length null components are not supported
```

I will update data as soon as I can reproduce the issue with a sample array.

**Comment by jpivarski** [comment]
Thanks! Knowing the layout of the array will be essential. The set of types that Arrow can write to Parquet has grown dramatically, and there used to be issues in which we left unreachable elements in the Arrow array that prevented it from being written to Parquet (instead of failing earlier), so we've seen things _like_ this before.

If you pickle your array and it's not too big, you can post it here instead of trying to make the original reproducible.

**Comment by drahnreb** [comment]
With the current version I can't reproduce the Error nor the array layout that caused it and cannot recover a pickled version of it. I close the issue.

**Comment by jpivarski** [comment]
I also encountered an Arrow bug when working with pyarrow 2.0 that went away in pyarrow 3.0. (It was an actual bug, though I don't remember the details.) Then, since I needed a LargeListArray fix in pyarrow 3.0, I made that the new minimum version for Awkward. It could be that increasing the Awkward version brought in a better pyarrow.
